samples/Ch18_using_libs/fig_18_14_pstl_usm_device.cpp

```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2023 Intel Corporation
+
+// SPDX-License-Identifier: MIT
+
+#include <oneapi/dpl/execution>
+#include <oneapi/dpl/algorithm>
+#include <sycl/sycl.hpp>
+
+int main() {
+  sycl::queue Q;
+  const int n = 10;
+  int * d_head = static_cast<int *>(
+    sycl::malloc_device(n * sizeof(int),
+                        Q.get_device(),
+                        Q.get_context()));
```

Comment:
This could be more concise by a) using the templated `malloc_device` and b) passing the queue instead of the device + context:

```suggestion
  int * d_head = sycl::malloc_device<int>(n, Q);
```